varcode/variant.py

```diff
@@ -223,9 +223,13 @@ def __setstate__(self, fields):
         self.ensembl = cached_release(fields.pop("release"))
         self.info = fields.pop("info", {})
 
-        # Remaining fields are simple properties that just get set.
+        # Remaining fields  are simple properties that just get set.
         self.__dict__.update(fields)
 
+        # TODO: switch to interbase coordinates so that 'end' becomes a
+        # mandatory rather than derived field
+        self.end = self.start + max(0, len(self.ref) - 1)
```

Comment:
Seems reasonable, but one problem is that the `__init__` method doesn't take an `end` argument. Maybe I should make `end` optional and only compute it from the trimmed ref if it's absent?